src/main/webapp/app/entities/iris/iris.model.ts

```diff
@@ -0,0 +1,45 @@
+import dayjs from 'dayjs';
+import { Exercise } from 'app/entities/exercise.model';
+import { User } from 'app/core/user/user.model';
+
+export class IrisConversation {
+    id: number;
+    programmingExercise?: Exercise;
+    user?: User;
+    latestMessageId?: string;
+    messageDescriptors: IrisMessageDescriptor[];
+    irisEnabled: boolean;
+}
+
+export enum IrisSender {
+    SERVER = 'server',
+    USER = 'user',
+}
+
+export class IrisServerMessageDescriptor {
+    messageId: number;
+    sender: IrisSender.SERVER;
+    messageContent?: IrisMessageContent;
+    sentAt: dayjs.Dayjs;
+    helpful?: boolean;
+}
+
+export class IrisClientMessageDescriptor {
+    sender: IrisSender.USER;
+    messageContent: IrisMessageContent;
+    messageId?: number;
```

Comment:
Needs to be the same as on the server.
```suggestion
    id?: number;
```
Also why is this in third place suddenly? You put the id on top everywhere else.